### PR TITLE
[#16]Test architecture setup

### PR DESCRIPTION
**Pull request by mattbudde**
### What's Changed?
As discussed in [16](https://github.com/belfastruby/belfastruby.github.io/issues/16)
- Added a few basic test scenarios (page has header etc..)
- Updated the readme with instructions on running the tests
- gemfile & gemfile.lock also updated appropriately

### CODE REVIEW

File: spec/spec_helper.rb [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+  Capybara.register_driver :selenium do |app|
+    # Configure selenium to use Chrome.
+    Capybara::Selenium::Driver.new(app, :browser => :chrome)
+  end
+
+  # Configure Capybara to load the website through rack-jekyll.
+  # (force_build: true) builds the site before the tests are run,
+  # so our tests are always running against the latest version
+  # of our jekyll site.
+  Capybara.app = Rack::Jekyll.new(force_build: true)
```
Comment (swmcc) [inline spec/spec_helper.rb]: `Capybara::Selenium::Driver.new(app, browser: :chrome)` - The community standards favour the JSON style.

File: spec/basic/render_spec.rb [hunk 2]
```diff
@@ -0,0 +1,20 @@
+require "spec_helper"
```
Comment (swmcc) [inline spec/basic/render_spec.rb]: I would put a `# frozen_string_literal: true` here.

File: spec/basic/render_spec.rb [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+describe "The Page renders", type: :feature, js: true do
+  before :each do
+    visit '/'
+  end
+
+  it "has a page title" do
+    expect(page).to have_content("A Community of Aspiring & Experienced Ruby Developers in Belfast")
+  end
+
+  it "has a header" do
```
Comment (swmcc) [inline spec/basic/render_spec.rb]: Community Standards - if there is no variable name then we favour using a single quote:

"Use doubles if you have to do this, ${name}"

or

'Use single if you have to this, Matt'

File: spec/basic/render_spec.rb (hunk 2 above)
Comment (mattbudde) [reply spec/basic/render_spec.rb]: About time i figured out what this does 👍

File: spec/spec_helper.rb (hunk 1 above)
Comment (mattbudde) [reply spec/spec_helper.rb]: Is there a particular style-guide I can take a look at?

Comment (swmcc) [reply spec/spec_helper.rb]: https://github.com/rubocop-hq/ruby-style-guide

Or `gem install rubocop` and then run rubocop -a on any file you have changed. Am sure you can automatically put it in your workflow with vscode https://github.com/misogi/vscode-ruby-rubocop might be worth a look.

**Pull request merged**